### ISSUE TITLE
IDE-TypeScript is causing unwanted modifications to open file on save

### ISSUE THREAD

**Issue by robertrossmann** [atom/ide-typescript#143]
After installing the 0.8.0 upgrade, when I save a JavaScript file (using _.mjs_ extension) the file is modified in an underisable way - an extra whitespace is inserted after each occurence of the _function bind (`::`)_ operator (ECMAScript proposal).

- Disabling `ide-typescript` resolves this problem
- This problem did not occur prior to 0.8.0 release.
- ESLint is not reporting any issues with this code so this is not a linter autofixer-related problem
- No _tsconfig.json_ or _jsconfig.json_ files are present in this project

#### Intended file contents

```js
// example.mjs
const map = Array.prototype.map
void [1, 2, 3]::map
```

#### Actual file contents after saving the file

```js
// example.mjs
const map = Array.prototype.map
void [1, 2, 3]:: map
//              ^ unintended extraneous space inserted
```

### Expected behaviour

`ide-typescript` does not modify original source files in any way unless explicitly requested (ie. via code actions etc.)

### Actual behaviour

`ide-typescript` modifies current file's contents by inserting unwanted whitespace

Actual, real-world project exhibiting this behaviour: https://github.com/strvcom/atlas.js/blob/master/packages/atlas/src/atlas.mjs

**Comment by damieng** [comment]
Glad you figured it out! :)

**Comment by zmb3** [comment]
I'm not sure this is working as intended.

I have other ide-xxx plugins that actually do support format on save, so I don't want to have to disable ide-ui's _format on save_ option only when working on typescript code.  However, if I leave it enabled I experience the issues described here.

If formatting isn't supported, shouldn't the format on save feature just be a no-op?
